WcaOnRails/app/models/competition.rb

```diff
@@ -305,6 +305,20 @@ def events
     eventSpecs.split.map { |e| Event.find_by_id(e.split("=")[0]) }.sort_by &:rank
   end
 
+  def has_event?(event)
+    event.id.in?(self.events.map(&:id))
+  end
+
+  def belongs_to_region?(region_or_country)
+    (self.countryId == region_or_country) || (self.country.continentId == region_or_country)
+  end
+
+  def search(search_param)
+    if search_param != nil
```

Comment:
It gets called with null when we just call /competitions.

This search works with reject on the controller. I reject anything that doesn't return true for the given search_param.